src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -405,9 +370,7 @@ HeapWord* ShenandoahFreeSet::try_allocate_in(ShenandoahHeapRegion* r, Shenandoah
   if (result != nullptr) {
     // Allocation successful, bump stats:
     if (req.is_mutator_alloc()) {
-      assert(req.is_young(), "Mutator allocations always come from young generation.");
-      generation->increase_used(size * HeapWordSize);
```

Comment:
Where is the generation's use being adjusted now?

Reply:
Waste, generation/heap used, bytes allocated and the concerns for the pacer are all consolidated into the overloaded `increase_used` method in `ShenandoahHeap`.